### PR TITLE
add user aws-observability

### PR DESCRIPTION
**Pull request by miyunari**
Added a dedicated user, who can run our service with restricted permissions. It's part of the `systemd-journal` group.

### CODE REVIEW

File: rhel-observability-aws-cloudwatch.spec [hunk 1]
```diff
@@ -24,6 +24,10 @@ mkdir -p %{buildroot}%{_unitdir}
 install -p -m 0644 -D %{SOURCE1} %{buildroot}%{_sysconfdir}/opentelemetry-collector-cloudwatch-config/config.yaml
 install -p -m 0644 -D %{SOURCE2} %{buildroot}%{_unitdir}/%{name}.service
 
+%pre
+/usr/bin/getent group aws-observability > /dev/null || /usr/sbin/groupadd -r aws-observability
+/usr/bin/getent passwd aws-observability > /dev/null || /usr/sbin/useradd -r -M -s /sbin/nologin -g aws-observability -G systemd-journal aws-observability
+
```
Comment (F-X64) [inline rhel-observability-aws-cloudwatch.spec:32]: Consider adding a %postun routine to clean the user once the package is removed from the system
```suggestion
%postun
/usr/sbin/userdel aws-observability
```

**Review by F-X64 — CHANGES_REQUESTED** [review]
We could go even further and add a require for the shadow-utils package to ensure that groupadd / useradd / userdel are available at ~~build~~ install time : 
```
  Requires(pre): shadow-utils
  Requires(postun): shadow-utils
```

**Comment by miyunari** [comment]
Thank you @FKolwa ! Makes sense to me :)  
I have one question is groupadd / useradd and userdel not only relevant at runtime?

**Comment by F-X64** [comment]
> Thank you @FKolwa ! Makes sense to me :) I have one question is groupadd / useradd and userdel not only relevant at runtime?

Yep you're absolutely right, thanks for picking up on that ;)

**Review by F-X64 — APPROVED** [review]
Looks good to me!

**Pull request merged**